dev/fattest.simplicity/test/componenttest/custom/junit/runner/CompoundFilterTests.java

```diff
@@ -0,0 +1,121 @@
+
+        @Test
+        public void dummyTestForRunner() {}
+    }
+
+    @WebServlet("*/")
+    public static class TestServletLiteMethod extends FATServlet {
+
+        @Mode(TestMode.LITE)
+        public void testMethod() {}
+    }
+
+    @Test
+    public void testMethodFilterSuperseedsTestClassFilter() throws Exception {
+        SyntheticServletTest testMethod = new SyntheticServletTest(TestServletLiteMethod.class, //
+                        TestClassFull.class.getField("NOOP"), //
+                        "FAKE/QUERY", //
+                        TestServletLiteMethod.class.getMethod("testMethod"));
+
+        //Replicate FATRunner.describeChild()
+        Description testDescription = Description.createTestDescription(TestClassFull.class, testMethod.getName(), testMethod.getAnnotations());
+
+        CompoundFilter filter = new CompoundFilter(testFiltersToApply);
+
+        assertTrue(filter.shouldRun(testDescription));
+    }
+
+    @Mode(TestMode.LITE)
+    @WebServlet("*/")
+    public static class TestServletLite extends FATServlet {
+
+        public void testMethod() {}
+    }
+
+    @Test
+    public void testServletFilterSuperseedsTestClassFilter() throws Exception {
+        SyntheticServletTest testMethod = new SyntheticServletTest(TestServletLite.class, //
+                        TestClassFull.class.getField("NOOP"), //
+                        "FAKE/QUERY", //
+                        TestServletLite.class.getMethod("testMethod"));
+
+        //Replicate FATRunner.describeChild()
+        Description testDescription = Description.createTestDescription(TestClassFull.class, testMethod.getName(), testMethod.getAnnotations());
+
+        CompoundFilter filter = new CompoundFilter(testFiltersToApply);
+
+        assertTrue(filter.shouldRun(testDescription));
+    }
+
+    @RunWith(FATRunner.class)
+    public static class TestClass {
+        public static LibertyServer NOOP;
+
+        @Test
+        public void dummyTestForRunner() {}
+    }
+
+    @Mode(TestMode.FULL) //Should be superseeded by method annotation
+    @WebServlet("*/")
+    public static class TestServletFullWithLiteMethod extends FATServlet {
+
+        @Mode(TestMode.LITE)
+        public void testMethod() {}
+    }
+
+    @Test
+    public void testServletMethodFilterSuperseedsTestServletFilter() throws Exception {
+        SyntheticServletTest testMethod = new SyntheticServletTest(TestServletFullWithLiteMethod.class, //
+                        TestClass.class.getField("NOOP"), //
+                        "FAKE/QUERY", //
+                        TestServletFullWithLiteMethod.class.getMethod("testMethod"));
+
+        //Replicate FATRunner.describeChild()
+        Description testDescription = Description.createTestDescription(TestClass.class, testMethod.getName(), testMethod.getAnnotations());
+
+        CompoundFilter filter = new CompoundFilter(testFiltersToApply);
+
+        assertTrue(filter.shouldRun(testDescription));
+    }
+
```

Comment:
Shouldn't there be some assertFalse tests for TestClass with a `@Mode(TestMode.FULL)` on the Servlet or Servlet method?  That was the whole point here.  We wanted to have the Test be lite mode, but the servlet be full mode.

Reply:
> We wanted to have the Test be lite mode, but the servlet be full mode.

That is the exact thing being tested in this case.  The servlet is is annotated with full mode, the test is annotated with lite mode. `fat.test.mode` is unset and thus we expect the method annotated with lite mode should run. 
I could add an additional test method and verify that an unannotated method does not run.

Reply:
Yeah this is what I wanted.  `testServletMethodFilterSupersedesTestServletFilter` validates that the liteMethod still runs with lite mode and defaultMethod is full mode because the servlet is marked full.  Without your change, it would have defaultMethod be lite mode because of the servlet annotation was ignored.